src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.FeatureCollection.cs

```diff
@@ -236,25 +235,7 @@ Stream IHttpResponseFeature.Body
             set => ResponseBody = value;
         }
 
-        PipeWriter IResponseBodyPipeFeature.Writer
-        {
-            get
-            {
-                if (!ReferenceEquals(_responseStreamInternal, ResponseBody))
-                {
-                    _responseStreamInternal = ResponseBody;
-                    ResponseBodyPipeWriter = PipeWriter.Create(ResponseBody, new StreamPipeWriterOptions(_context.MemoryPool));
```

Comment:
Only downside here is that the adapter used Kestrel's pool when it could (though that could still be implemented since it's a feature 🤔 )